topdown/encoding_test.go

```diff
@@ -0,0 +1,139 @@
+package topdown
```

Comment:
Given that you already added tests cases there, https://github.com/open-policy-agent/opa/pull/2867/files#diff-890ef04c809dedbc709f0213d9d5bf9dc536cc804fa4d2e42ae58e52cb90c8f5R1, what do these tests give us beyond those other ones? 🤔 I think the yaml test cases are the preferred way to do this now.

Reply:
I initially had some problems to newly created yaml tests. But it works now, I've moved all tests to yaml and removed encoding_test.go.

Reply:
@srenatus Not sure I understand codecov/patch, but it seem it want to cover the builtins.StringOperand failure:
https://codecov.io/gh/open-policy-agent/opa/compare/cd8cb1d7e4eb4b05edb50c3a81ca02f2ef10e8ed...cdc18f0ad34519a7bc79e6e0ba278759c377de08/diff

Reply:
Yeah, there's two ways that invalid arguments are checked: at the rego level, and at evaluation time. Those tests expect the module code to be parsable as rego modules, and failures like `x := base64url.encode_no_pad(123)` are **not**, since it's all static, it fails with `rego_type_error`.

What I've done before to trigger the eval time error is to use data. In this case, I believe this covers the bad operand case:
```yaml
- note: base64urlbuiltins/encode-2 without padding/operand error
  data:
    operand: 120
  query: data.generated.p = x
  modules:
  - |
    package generated
    p = x {
      base64url.encode_no_pad(data.operand, x)
    }
  want_error: "test-0.rego:3: eval_type_error: base64url.encode_no_pad: operand 1 must be string but got number"
  want_error_code: eval_type_error
```

Reply:
BTW this is how you'd do it. I'm not sure we strive for coverage of those lines, though; it may be fine to follow the existing builtins' examples. It's still humans reviewing, and the coverage check can be overridden.

Reply:
Thanks for the example. I agree it doesn't seem that valuable.